### DISCUSSION TITLE
Godror and 32-bit

### DISCUSSION THREAD

**Discussion by Nappy-Says** [godror/godror#246] [Q&A] [answered]
Hello! a question arose on the use of godror with oracle **10g 32-bit**.
On my PC there is a 32 bit version of Oracle. when you run the golang application, everything collapses with the words:
`panic: ORA-00000: DPI-1047: Cannot locate a 64-bit Oracle Client library: "..\oracle\bin\oci.dll is not the correct architecture". See https://oracle.github.io/odpi/doc/installation.html#windows for help.`

Is it possible to set Godror to a different bitness?

**Comment by sudarshan12s** [comment, marked as answer]
I think after setting proper  env
```
CGO_ENABLED = 1
GOOS = windows
GOARCH = 386
```

`go build`

 should generate a 32-bit binary. 

cross compilers for C are also required

https://github.com/godror/godror/blob/main/doc/cross-compile.md#zig
